ctru-rs/src/gfx.rs

```diff
@@ -114,14 +114,12 @@ impl Gfx {
     ///
     /// Note that the pointer returned by this function can change after each call to this function
     /// if double buffering is enabled
-    pub fn get_raw_framebuffer(&self, screen: Screen, side: Side) -> (*mut u8, u16, u16) {
-        unsafe {
-            let mut width: u16 = 0;
-            let mut height: u16 = 0;
-            let buf: *mut u8 =
-                ctru_sys::gfxGetFramebuffer(screen.into(), side.into(), &mut width, &mut height);
-            (buf, width, height)
-        }
+    pub unsafe fn get_raw_framebuffer(&self, screen: Screen, side: Side) -> (*mut u8, u16, u16) {
```

Comment:
Clippy warns that this should have a `# Safety` section to note why it's unsafe.

Reply:
I should run clippy more. I will add it when changing the other stuff, though it's pretty obvious that a getter for raw pointers to a framebuffer (which as stated in libctru, it may not even be the framebuffer currently used if you are in double buffering mode), should be unsafe.

Reply:
Well actually it's safe to do that. What's unsafe is dereferencing the pointer. This program is perfectly safe:
```rust
fn foo(foo: *mut usize) {
    println!("Hello, I'm at {:?}", foo);
}

fn main() {
    foo(0 as *mut usize);
}
```
[Playground link](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=d5c688c8afb983fe1e9b8dbe794e9a9b)

Reply:
An "unsafe" function doesn't have to run unsafe code. I marked it unsafe because it returns a pointer to raw graphic memory, which isn't something safe to do under any circumstance.

Reply:
It is safe to return a raw pointer to somewhere. See the stuff I linked earlier

Reply:
As long as there's no way to trigger undefined behavior it's safe (just need to check that the libctru call is safe under the hood).